janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestElasticSearchClient.java

```diff
@@ -356,17 +361,22 @@ public void deleteIndex(String indexName) throws IOException {
     public void bulkRequest(List<ElasticSearchMutation> requests, String ingestPipeline) throws IOException {
         final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
         for (final ElasticSearchMutation request : requests) {
-            final Map<String, String> requestData;
-            if(useMappingTypes){
-                requestData = ImmutableMap.of(
+            Map<String, Object> requestData = new HashMap<>();
+            if (useMappingTypes) {
+                requestData.putAll(ImmutableMap.of(
```

Comment:
You create additional Immutable map which is redundant.
You can just use `put` method for each entry instead of creating additional map and then using putAll.